packages/protocol-sdk/src/1155-create-helper.ts

```diff
@@ -0,0 +1,319 @@
+  maxSupply?: bigint | number;
+  createReferral?: Address;
+  nextTokenId: bigint;
+  mintToCreatorCount: bigint | number;
+  account: Address;
+  tokenMetadataURI: string;
+  fixedPriceMinterAddress: Address;
+  salesConfig: SalesConfigParamsType;
+  royaltySettings?: RoyaltySettingsType;
+}) {
+  if (!maxSupply) {
+    maxSupply = OPEN_EDITION_MINT_SIZE;
+  }
+  maxSupply = BigInt(maxSupply);
+  mintToCreatorCount = BigInt(mintToCreatorCount);
+
+  const salesConfigWithDefaults = Object.assign(
+    // Set static sales default.
+    DEFAULT_SALE_SETTINGS,
+    // Override with user settings.
+    salesConfig,
+  );
+
+  const setupActions = [
+    encodeFunctionData({
+      abi: zoraCreator1155ImplABI,
+      functionName: "addPermission",
+      args: [0n, fixedPriceMinterAddress, PERMISSION_BIT_MINTER],
+    }),
+    encodeFunctionData({
+      abi: zoraCreator1155ImplABI,
+      functionName: "assumeLastTokenIdMatches",
+      args: [nextTokenId - 1n],
+    }),
+    createReferral
+      ? encodeFunctionData({
+          abi: zoraCreator1155ImplABI,
+          functionName: "setupNewTokenWithCreateReferral",
+          args: [tokenMetadataURI, maxSupply, createReferral],
+        })
+      : encodeFunctionData({
+          abi: zoraCreator1155ImplABI,
+          functionName: "setupNewToken",
+          args: [tokenMetadataURI, maxSupply],
+        }),
+    encodeFunctionData({
+      abi: zoraCreator1155ImplABI,
+      functionName: "callSale",
+      args: [
+        nextTokenId,
+        fixedPriceMinterAddress,
+        encodeFunctionData({
+          abi: zoraCreatorFixedPriceSaleStrategyABI,
+          functionName: "setSale",
+          args: [nextTokenId, salesConfigWithDefaults],
+        }),
+      ],
+    }),
+  ];
+
+  if (mintToCreatorCount) {
+    setupActions.push(
+      encodeFunctionData({
+        abi: zoraCreator1155ImplABI,
+        functionName: "adminMint",
+        args: [account, nextTokenId, mintToCreatorCount, "0x"],
+      }),
+    );
+  }
+
+  if (royaltySettings) {
+    setupActions.push(
+      encodeFunctionData({
+        abi: zoraCreator1155ImplABI,
+        functionName: "updateRoyaltiesForToken",
+        args: [
+          nextTokenId,
+          {
+            royaltyMintSchedule: 0,
+            royaltyBPS: royaltySettings?.royaltyBPS || royaltyBPSDefault,
+            royaltyRecipient: royaltySettings?.royaltyRecipient || account,
+          },
+        ],
+      }),
+    );
+  }
+
+  return setupActions;
+}
+
+export const getTokenIdFromCreateReceipt = (
+  receipt: TransactionReceipt,
+): bigint | undefined => {
+  for (const data of receipt.logs) {
+    try {
+      const decodedLog = decodeEventLog({
+        abi: zoraCreator1155ImplABI,
+        eventName: "SetupNewToken",
+        ...data,
+      });
+      if (decodedLog && decodedLog.eventName === "SetupNewToken") {
+        return decodedLog.args.tokenId;
+      }
+    } catch (err: any) {}
+  }
+};
+
+async function getContractExists(
```

Comment:
i think it may be cleaner to just break this into two functions:

```ts
const getDeterminsticContractExists({contractConfig})

const getContractExists({address})
```